### ISSUE TITLE
Accessing nested properties does not work (JSON)

### ISSUE THREAD

**Issue by Tarpsvo** [whitecube/nova-flexible-content#13]
If I'm trying to access a field's value that is inside a JSON (ie `data->title`, where `data` is JSON), the resolver fails to retrieve the value. More specifically, the `extractValueFromResource` function.

I propose that `extractValueFromResource` should work similary to Nova's `Field` `resolveAttribute`.

**Comment by voidgraphics** [comment]
Could you give a more practical example to allow us to easily reproduce the issue? That would help us out a bunch.

Thanks!

**Comment by Tarpsvo** [comment]
I'll walk you through what happened, but I don't have the time right now to make an example project, sorry.

Let's say I have a model with a `data` JSON attribute and I want to save a `Flexible` field inside the `data`, for example in a property `data->banners`.

The model has Eloquent `$casts` defined as follows: `['data' => 'object'];`

Inside [Resolver.php](https://github.com/whitecube/nova-flexible-content/blob/0d471f154364e19471f0424f48daa80f3d2ea8c5/src/Value/Resolver.php#L57), the accessor `$resource->$attribute` call returns null when `$attribute` is `data->banners`, even though the data is there. I tried wrapping the accessor in curly braces (`$resource->{$attribute}`), but that didn't help. When accessing the value directly, not through the string, the value is retrieved as expected (just doing `$resource->data->banners`).

However, when I tried creating regular fields instead of Flexible fields, I noticed that they worked when nested in JSON, so I quickly glanced at the code inside Nova's `Field.php#208`:

```php
protected function resolveAttribute($resource, $attribute)
{
    return data_get($resource, str_replace('->', '.', $attribute));
}
```

I propose that Flexible's `Resolver` should also use `data_get` as it fixes the issue of accessing nested properties.

```php
protected function extractValueFromResource($resource, $attribute)
{
    return data_get($resource, str_replace('->', '.', $attribute));
}
```

NB! `Collection` expects the values to be object, so the Eloquent `$casts` has to definitely cast the JSON property into an object (not an array).

**Comment by voidgraphics** [comment]
Thanks for the more in-depth explanation and the suggested solution. We'll look into it shortly!

**Comment by Tarpsvo** [comment]
Thanks for your attention!

**Comment by jankremlacek** [comment]
hi @Tarpsvo, it seems a bit more complex with the Resolver. Here is an example:

`Whitecube\NovaFlexibleContent\Value\Resolver.php` line: 39

`if(!is_array($value)) $value = json_decode($value);`

It assumes there is no `$casts` on the JSON field in the model, or `cast` to `object`.
The JSON field can be in Eloquent casted to `array` as well, which produces assoc array and the rest of resolving code would not work (_as it accesses its nested values as properties_).

Would like to PR with improvements, just let me know how you wanna approach it. We can check the `$value` first by `is_object` instead of `is_array`, then by `is_array` and convert it to object, like for example:

```php
if (!is_object($value)) {
    $value = is_array($value)
        ? (object) $value
        : json_decode($value);
}
```

**Comment by Tarpsvo** [comment]
See PR #18. Ran some basic tests and it worked as expected in both the usual usecases, when data was nested in JSON and when data was cast to either `object` or `array`.

**Comment by voidgraphics** [comment]
This is included in the new [v0.1.3](https://github.com/whitecube/nova-flexible-content/releases/tag/v0.1.3) release.
